climakitae/timeseriestools.py

```diff
@@ -0,0 +1,222 @@
+import xarray as xr
+import param
+import panel as pn
+import hvplot.xarray
+import datetime as dt
+import dask
+import tempfile, os
+
+
+class TimeSeriesParams(param.Parameterized):
+    """
+    An object to hold time-series parameters, which depends only on the 'param' library.
+    Currently used in '_timeseries_visualize', which uses 'panel' to draw the gui, but another
+    UI could in principle be used to update these parameters instead.
+    """
+
+    def __init__(self, dataset, **params):
+        super().__init__(**params)
+        self.data = dataset
+
+    anomaly = param.Boolean(default=True, label="Difference from a historical mean")
+    reference_range = param.CalendarDateRange(
+        default=(dt.datetime(1980, 1, 1), dt.datetime(2012, 12, 31)),
+        bounds=(dt.datetime(1980, 1, 1), dt.datetime(2021, 12, 31)),
+    )
+    remove_seasonal_cycle = param.Boolean(default=False)
+    smoothing = param.ObjectSelector(default="None", objects=["None", "running mean"])
+    _time_scales = dict(
+        [("hours", "H"), ("days", "D"), ("months", "MS"), ("years", "AS-SEP")]
+    )
+    num_timesteps = param.Integer(default=0, bounds=(0, 240))
+    separate_seasons = param.Boolean(
+        default=False, label="Disaggregate into four seasons"
+    )
+
+    extremes = param.ObjectSelector(
+        default="None", objects=["None", "min", "max", "percentile"]
+    )
+    resample_window = param.Integer(default=1, bounds=(1, 30))
+    resample_period = param.ObjectSelector(default="A", objects=_time_scales)
```

Comment:
this is throwing an error:
`from climakitae import timeseriestools as tst`
...
`ValueError: A not in parameter's list of possible objects, valid options include [H, D, MS, AS-SEP]`

Reply:
Whoops -- was in too much of a hurry with that last change. Fixed.